### PR TITLE
[Fix/UnitTest] Add mutex to count number of callbacks @open sesame 12/14 18:32

### PR DESCRIPTION
**Pull request by dongju-chae**
This patch adds mutex to count number of callbacks.

It resolves https://github.com/nnstreamer/nnstreamer/issues/2941

Signed-off-by: Dongju Chae <dongju.chae@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

### CODE REVIEW

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc [hunk 1]
```diff
@@ -680,13 +680,20 @@ TEST (nnstreamer_nnfw_mlapi, invoke_single_02_n)
...
-    while (*sink_called_cnt < expected_cnt) {       \
-      g_usleep (TEST_DEFAULT_SLEEP_TIME);           \
+    while (1) {                                     \
+      gboolean exit = FALSE;                        \
+      g_mutex_lock (&g_test_mutex);                 \
+      exit = (*sink_called_cnt >= expected_cnt);    \
       waiting_time += TEST_DEFAULT_SLEEP_TIME;      \
+      g_mutex_unlock (&g_test_mutex);               \
+      if (exit) break;                              \
+      g_usleep (TEST_DEFAULT_SLEEP_TIME);           \
```
Comment (myungjoo) [inline tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc:690]: Do not rely on assert for behaviors. It should behave correctly even ASSERT is ignored.

Comment (dongju-chae) [reply tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc]: `ASSERT_RT` was removed. Thanks.

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc [hunk 2]
```diff
@@ -677,15 +677,31 @@ TEST (nnstreamer_nnfw_mlapi, invoke_single_02_n)
...
+static gboolean
+wait_for_sink (guint expected_cnt)
+{
+  guint waiting_time = 0;
+  gboolean done = FALSE;
+
+  while (!done && waiting_time < TEST_TIMEOUT_LIMIT) {
+    g_mutex_lock (&g_test_mutex);
+
+    done = (*sink_called_cnt >= expected_cnt);
```
Comment (again4you) [inline tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc:694]: Compile time error :)


```bash
[   43s] ../tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc:694:14: error: 'sink_called_cnt' was not declared in this scope

```

Comment (dongju-chae) [reply tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc]: That's right :) I updated the PR. Thanks.

**Review by gichan-jang — APPROVED** [review]
LGTM

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc (hunk 1 above)
Comment (myungjoo) [reply tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc]: No. Do not remove it. It's a valid assertion.

You need to make it work even if ASSERT is ignored by compilers.

**Review by again4you — APPROVED** [review]
LGTM!

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc (hunk 1 above)
Comment (dongju-chae) [reply tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cc]: Got it. I'll change `EXPECT_TRUE` to `ASSERT_TRUE`. The tests will work without this assert.

**Review by myungjoo — APPROVED** [review]

**Pull request merged**